config/hocon-mp/src/main/java/io/helidon/config/hocon/mp/HoconMpConfigSource.java

```diff
@@ -0,0 +1,401 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.config.hocon.mp;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import io.helidon.config.ConfigException;
+import io.helidon.config.MutabilitySupport;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigList;
+import com.typesafe.config.ConfigObject;
+import com.typesafe.config.ConfigParseOptions;
+import com.typesafe.config.ConfigResolveOptions;
+import com.typesafe.config.ConfigValue;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+/**
+ * MicroProfile {@link org.eclipse.microprofile.config.spi.ConfigSource} that can be used
+ * to add HOCON/JSON files from classpath or file system using the
+ * {@link org.eclipse.microprofile.config.spi.ConfigProviderResolver#getBuilder()}.
+ * <p>The HOCON/JSON file is transformed to a flat map as follows:</p>
+ * <strong>Object nodes</strong>
+ * <p>
+ * Each node in the tree is dot separated.
+ * <pre>
+ * server = {
+ *   host = "localhost"
+ *   port= 8080
+ * }
+ * </pre>
+ * Will be transformed to the following properties:
+ * <pre>
+ * server.host=localhost
+ * server.port=8080
+ * </pre>
+ * <strong>List nodes (arrays)</strong>
+ * <p>
+ * Each node will be indexed (0 based)
+ * <pre>
+ * providers =
+ *   [{abac = {enabled = true}}]
+ * names = [
+ *   first
+ *   second
+ *   third
+ * ]
+ * </pre>
+ * Will be transformed to the following properties:
+ * <pre>
+ * providers.0.abac.enabled=true
+ * names.0=first
+ * names.1=second
+ * names.2=third
+ * </pre>
+ */
+@SuppressWarnings("rawtypes")
+public class HoconMpConfigSource implements ConfigSource {
+    private final Map<String, String> properties;
+    private final String name;
+
+    private static final boolean RESOLVING_ENABLED = true;
+    private static final ConfigResolveOptions RESOLVE_OPTIONS = ConfigResolveOptions.defaults();
+    private static final HoconMpConfigIncluder INCLUDER = new HoconMpConfigIncluder();
+    private static final ConfigParseOptions PARSE_OPTIONS = ConfigParseOptions.defaults().appendIncluder(INCLUDER);
```

Comment:
see above - if we do not use static includler, we should not use static parse options.

Reply:
Same as above

Reply:
done